RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -0,0 +1,318 @@
+#ifndef DAClusterizerInZT_vect_h
+#define DAClusterizerInZT_vect_h
+
+/**\class DAClusterizerInZT_vect
+
+ Description: separates event tracks into clusters along the beam line
+
+	Version which auto-vectorizes with gcc 4.6 or newer
+
+ */
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+#include <memory>
+#include <stdlib.h>
+#include <malloc.h>
+
+template <typename T, std::size_t N = 16>
+class AlignmentAllocator {
```

Comment:
From what I can tell, the C++ standard has added the keywork `alignas` to C++11 in order to replace the compiler specific `__attribute__ ((aligned (16)))`. However, I do not believe `alignas` would apply to the value pointed, but by the actual member data item itself. 

The best I can tell, the C++11 standard wants one to use a 'dummy type' (using either `std::aligned_union` or `std::aligned_storage`) for the pointer type and for the call to `new` to create the initial memory area and then use placement new to insert the objects into the already allocated memory area.

It might also just be easiest to create a POD like
```
struct alignas(16) AlignedDouble { 
   double value_; 
   double& operator double() { return value_;} 
;
```
and then do
```
AlignedDouble * __restrict__ _z;
```

Then passing an `AlignedDouble` to a function means the compiler will know its alignment and can better auto-vectorize since it is guaranteed (by the type system) to be aligned.